specification/cognitiveservices/data-plane/AnomalyDetector/stable/v1.1/cadl/multivariate/models.cadl

```diff
@@ -0,0 +1,360 @@
+import "@cadl-lang/rest";
+import "@azure-tools/cadl-azure-core";
+
+using Cadl.Http;
+using Cadl.Rest;
+using Azure.Core;
+
+namespace AnomalyDetector.Multivariate;
+
+enum FillNAMethodKnownValues {
+  "Previous",
+  "Subsequent",
+  "Linear",
+  "Zero",
+  "Fixed",
+}
+
+@knownValues(FillNAMethodKnownValues)
+model FillNAMethod is string;
+
+enum DetectionStatus {
+  Created: "CREATED",
+  Running: "RUNNING",
+  Ready: "READY",
+  Failed: "FAILED",
+}
+
+enum DataSchema {
+  "OneTable",
+  "MultiTable",
+}
+
+enum AlignMode {
+  "Inner",
+  "Outer",
+}
+
+enum ModelStatus {
+  Created: "CREATED",
+  Running: "RUNNING",
+  Ready: "READY",
+  Failed: "FAILED",
+}
+
+@doc("Detection results for the given resultId.")
+@resource("multivariate/detect-batch")
+model DetectionResult {
+  @doc("Result identifier, which is used to fetch the results of an inference call.")
+  @key
+  @format("uuid")
+  resultId: string;
+
+  @doc("Multivariate anomaly detection status.")
+  summary: DetectionResultSummary;
+
+  @doc("Detection result for each timestamp.")
+  results: AnomalyState[];
+}
+
+@doc("Multivariate anomaly detection status.")
+model DetectionResultSummary {
+  @doc("Status of detection results. One of CREATED, RUNNING, READY, and FAILED.")
+  status: DetectionStatus;
+
+  @doc("Error message when detection is failed.")
+  errors?: ErrorResponse[];
+  variableStates?: VariableState[];
+
+  @doc("""
+Detection request for batch inference. This is an asynchronous inference which
+will need another API to get detection results.
+""")
+  setupInfo: DetectionRequest;
+}
+
+@error
+model ErrorResponse {
+  @header
+  @doc("Error code.")
+  "x-ms-error-code"?: string,
+
+  @doc("The error code.")
+  code: string;
+
+  @doc("The message explaining the error reported by the service.")
+  message: string;
+}
+
+model VariableState {
```

Comment:
I think we should have description for all models and properties, is that correct?

Reply:
Yes, @jr-MS will help with this